### PR TITLE
Enable range queries on datetime and u64 fields

### PR DESCRIPTION
**Pull request by guilload**
### Description
Per title. I need this feature for the Jaeger integration (filters on span start timestamp or duration).

### How was this PR tested?
Added unit tests.

### CODE REVIEW

**Comment by guilload** [comment]
This is ready for review.

**Comment by fulmicoton** [comment]
@PSeitz can you review?

**Comment by guilload** [comment]
@PSeitz, PTAL!

File: quickwit/quickwit-doc-mapper/src/query_builder.rs [hunk 1]
```diff
@@ -126,19 +126,23 @@ fn extract_field_name(leaf: &UserInputLeaf) -> Option<&str> {
 
 fn is_valid_field_for_range(field_entry: &FieldEntry) -> anyhow::Result<()> {
     match field_entry.field_type() {
-        FieldType::IpAddr(ip_addr_options) => {
-            if !ip_addr_options.is_fast() {
+        FieldType::Date(_)
+        | FieldType::IpAddr(_)
+        | FieldType::F64(_)
+        | FieldType::I64(_)
+        | FieldType::U64(_) => {
```
Comment (PSeitz) [inline quickwit/quickwit-doc-mapper/src/query_builder.rs:133]: `Bool` is also supported

File: quickwit/quickwit-doc-mapper/src/query_builder.rs [hunk 2]
```diff
@@ -126,19 +126,23 @@ fn extract_field_name(leaf: &UserInputLeaf) -> Option<&str> {
...
                     field_entry.name()
                 );
             }
         }
         other_type => {
             anyhow::bail!(
-                "`{}` is of type `{:?}`. Range queries are only supported for IP Field at the \
-                 moment.",
+                "Field `{}` is of type `{:?}`. Range queries are only supported on datetime, IP, \
+                 and numeric fields at the moment.",
```
Comment (PSeitz) [inline quickwit/quickwit-doc-mapper/src/query_builder.rs:145]: and bool

**Pull request merged**